src/server/req_delete.c

```diff
@@ -201,30 +201,30 @@ acct_del_write(char *jid, job *pjob, struct batch_request *preq, int nomail)
  *		done by req_deletejob()
  *
  * @param[in]	preq	- Batch request structure.
+ * @param[in]	pjob	- pointer to job
+ * @param[in]	parent	- pointer to parent job for subjob
+ * @param[in]	jt	- job type
  *
  * @return	int
  * @retval	TRUE	- Job history  has been purged
  * @retval	FALSE	- Job is not a history job
  */
 int
-check_deletehistoryjob(struct batch_request * preq)
+check_deletehistoryjob(struct batch_request * preq, job *pjob, job *parent, int jt)
 {
-	job *histpjob;
-	job *pjob;
-	int historyjob;
-	int histerr;
 	char *jid;
+	job *psjob;
+	char *sjid;
+	int i;
+
 	jid = preq->rq_ind.rq_delete.rq_objname;
 
-	histpjob = find_job(jid);
 	/*
-	 * if histpjob returns NULL then we can be sure its a array subjob or a range of subjobs . If the array
-	 * subjob or range of subjobs are in a history state then reject the request as we cant delete history
-	 * of array subjobs
+	 * If the array subjob or range of subjobs are in a history state then reject the request
+	 * as we cant delete history of array subjobs
```

Comment:
Please rephrase the comment as we can now actually delete the history for subjobs as well.

Reply:
Its reverted in rc 20 https://github.com/PBSPro/pbspro/pull/590/commits/b01fcac3e94363966f635e6fad726a5342288a79